### PR TITLE
[YouTube QA] Responsive graph

### LINKED ISSUE #7014: [YouTube QA] "Types of Regreted Content" graph should scale at smaller widths
- [x] Graph should scale down at narrow widths (bar lengths reduce)
- [x] Dark red color should be at the top (not at the bottom for Fake Engagement)
~Titles move above each bar~

Screen cap from mock up
image

Screen cap from staging
image

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #7014 #7036

Prettier made most of the changes. I'll self-review to help determine what was actually changed vs. what was prettier doing its thing. 

**Link to sample test page**: https://foundation-s-7014-respo-udp2rn.herokuapp.com/en/campaigns/regrets-reporter/findings/#what-we-found

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/categories.html [hunk 1]
```diff
@@ -17,8 +17,8 @@ <h2 class="heading-1">{% trans "The most common reported category was misinforma
         <div class="col-12 col-md-8 categories__chart-heading">
           <h2 class="heading-5 mb-2">{% trans "Types of Regretted Content" %}</h2>
         </div>
-        <div class="col-12 categories__chart-wrap">
-          <canvas class="categories__chart" width="930" height="450" id="categories-bar-chart"></canvas>
+        <div class="col-12">
+          <canvas width="930" height="450" id="categories-bar-chart"></canvas>
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube-regrets-2021/fragments/categories.html:21]: I removed the old styling

File: source/js/foundation/pages/youtube-regrets/categories-bar-chart.js [hunk 2]
```diff
@@ -8,11 +8,11 @@ export const initYouTubeRegretsCategoriesBarChart = () => {
     gettext("VIOLENT OR GRAPHIC CONTENT"),
     gettext("COVID-19 MISINFORMATION"),
     gettext("HATE SPEECH"),
-    gettext("SPAM"),
+    gettext("SPAM, DECEPTIVE PRACTICES"),
     gettext("NUDITY & SEXUAL CONTENT"),
     gettext("CHILD SAFETY"),
     gettext("HARASSMENT & CYBERBULLYING"),
-    gettext("HARMFUL OR DANGEROUS"),
+    gettext("HARMFUL OR DANGEROUS CONTENT"),
```
Comment (KalobTaulien) [inline source/js/foundation/pages/youtube-regrets/categories-bar-chart.js:15]: New content. @TheoChevalier you might need to know about this.

File: source/js/foundation/pages/youtube-regrets/categories-bar-chart.js [hunk 3]
```diff
@@ -40,23 +40,26 @@ export const initYouTubeRegretsCategoriesBarChart = () => {
...
             "#404040",
             "#2D2E7A",
             "#4345B6",
             "#595CF3",
-            "#A0A2F8",
-            "#E7E7FC",
-            "#FFFFFF",
+            "#8F8BF0",
+            "#ABADFC",
+            "#E3E3FC",
```
Comment (KalobTaulien) [inline source/js/foundation/pages/youtube-regrets/categories-bar-chart.js:54]: Proper colors. Looks like we were missing at least one as well.

File: source/js/foundation/pages/youtube-regrets/categories-bar-chart.js [hunk 4]
```diff
@@ -86,9 +89,14 @@ export const initYouTubeRegretsCategoriesBarChart = () => {
...
-              size: 12,
-              family: "Nunito Sans",
+            font: function (context) {
+              var width = context.chart.width;
+              const size = width >= 410 ? 12 : 8;
+
+              return {
+                family: "Nunito Sans",
+                size: size,
+              };
```
Comment (KalobTaulien) [inline source/js/foundation/pages/youtube-regrets/categories-bar-chart.js:99]: Variable font sizing for the y-axis labels. It'll make the labels smaller so they aren't cut off by the viewport on small devices.

File: source/sass/views/youtube-regrets-2021.scss [hunk 5]
```diff
@@ -253,14 +253,13 @@
...
-        padding: 0;
-        width: 100%;
-        overflow-x: auto;
+    @media (max-width: 480px) {
+      canvas {
+        /* stylelint-disable */
+        // Very important we overwrite what Chart.js is producing in the [style] attribute on mobile
+        // specifically for, and exactly at, 480px
+        height: 430px !important;
+        /* stylelint-enable */
```
Comment (KalobTaulien) [inline source/sass/views/youtube-regrets-2021.scss:262]: Chart.js was forcing a chart height through the style='' attribute. We had to overwrite it specifically at 480px and smaller.

**Review by KalobTaulien — COMMENTED** [review]
Stuff below is what I changed. The rest was prettier making code prettier than it used to be.

**Review by kristinashu — APPROVED** [review]
Design looks great, thank you! Ready for code review.

File: source/js/foundation/pages/youtube-regrets/categories-bar-chart.js (hunk 2 above)
Comment (TheoChevalier) [reply source/js/foundation/pages/youtube-regrets/categories-bar-chart.js:15]: Thanks for the heads up! Got translation going for these

**Review by danielfmiranda — APPROVED** [review]
Thanks for tagging me on this Kalob! 

It was really interesting to see how they implemented the charts on the youtube regrets page.

Also gave the changes a look, and they look good to me! Same for testing the functionality on the page. Approving!

**Pull request merged**